packages/Search/test/tstBoostGeometryAdapters.cpp

```diff
@@ -0,0 +1,248 @@
+    TEST_EQUALITY( bg::distance( unit_box, p ), 0. );
+
+    p = {{-1., -1., -1.}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), std::sqrt( 3. ) );
+    TEST_EQUALITY( bg::distance( p, unit_box ), std::sqrt( 3. ) );
+
+    p = {{-1., .5, -1.}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), std::sqrt( 2. ) );
+    TEST_EQUALITY( bg::distance( p, unit_box ), std::sqrt( 2. ) );
+
+    p = {{-1., .5, .5}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), 1. );
+    TEST_EQUALITY( bg::distance( p, unit_box ), 1. );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, expand )
+{
+    using dtk::equals;
+    dtk::Box box;
+    // NOTE even though not considered valid, default constructed DTK box can be
+    // expanded using Boost.Geometry algorithm.
+    TEST_ASSERT( !bg::is_valid( box ) );
+    bg::expand( box, dtk::Point{{0., 0., 0.}} );
+    dtk::expand( box, {{1., 1., 1.}} );
+    TEST_ASSERT( equals( box, {{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    bg::expand( box, dtk::Box{{{1., 2., 3.}}, {{4., 5., 6.}}} );
+    dtk::expand( box, {{{-1., -2., -3.}}, {{0., 0., 0.}}} );
+    TEST_ASSERT( equals( box, {{{-1., -2., -3.}}, {{4., 5., 6.}}} ) );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, centroid )
+{
+    using dtk::equals;
+    // For convenience define a function that returns the centroid.
+    // Boost.Geometry defines both `void centroid(Geometry const & geometry,
+    // Point &c )` and `Point return_centroid(Geometry const& geometry)`
+    auto const dtkReturnCentroid = []( dtk::Box b ) {
+        dtk::Point c;
+        dtk::centroid( b, c );
+        return c;
+    };
+
+    // Interestingly enough, even though for Boost.Geometry, the DTK default
+    // constructed "empty" box is not valid, it will still calculate its
+    // centroid.  Admittedly, the result (centroid at origin) is garbage :)
+    dtk::Box empty_box = {};
+    TEST_ASSERT( !bg::is_valid( empty_box ) );
+    TEST_ASSERT( equals( bg::return_centroid<dtk::Point>( empty_box ),
+                         {{0., 0., 0.}} ) );
+    TEST_ASSERT( equals( dtkReturnCentroid( empty_box ), {{0., 0., 0.}} ) );
+
+    dtk::Box unit_box = {{{0., 0., 0.}}, {{1., 1., 1.}}};
+    TEST_ASSERT(
+        equals( bg::return_centroid<dtk::Point>( unit_box ), {{.5, .5, .5}} ) );
+    TEST_ASSERT( equals( dtkReturnCentroid( unit_box ), {{.5, .5, .5}} ) );
+
+    // NOTE DTK does not have an overload of centroid() for Point at the
+    // moment.
+    dtk::Point a_point = {{1., 2., 3.}};
+    TEST_ASSERT(
+        equals( bg::return_centroid<dtk::Point>( a_point ), a_point ) );
+    // TEST_ASSERT( equals( dtk::centroid(
+    //     []( dtk::Point p ) {
+    //         dtk::Point c;
+    //         dtk::centroid( p, c );
+    //         return c;
+    //     }( a_point ),
+    //     a_point ) ) );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, is_valid )
+{
+    // NOTE "empty" box is considered as valid in DataTransferKit but it is
+    // not according to Boost.Geometry
```

Comment:
What's the difference in how they determine validity?

Reply:
Look at the messages below.  Boost.Geometry claims to follow the [OGC](http://www.opengeospatial.org) standards on geometry.